### PR TITLE
Update lakeFSFS to use (upcoming) lakeFS SDK 1.0

### LINKED ISSUE #6678: Refactor lakeFSFS to use lakeFS 1.0 SDK-to-be

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #6678.

Uses the Java SDK being added in #6656 .

### CODE REVIEW

File: clients/hadoopfs/pom.xml [hunk 1]
```diff
@@ -276,8 +276,8 @@ To export to S3:
     <dependencies>
         <dependency>
             <groupId>io.lakefs</groupId>
-            <artifactId>api-client</artifactId>
-            <version>0.91.0</version>
+            <artifactId>sdk-client</artifactId>
+            <version>0.1.0-SNAPSHOT</version>
```
Comment (Jonathan-Rosenberg) [inline clients/hadoopfs/pom.xml:280]: I wonder if this was supposed to be checked but probably better to find out

File: Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ CLIENT_JARS_BUCKET="s3://treeverse-clients-us-east/"
 # https://openapi-generator.tech
 OPENAPI_LEGACY_GENERATOR_IMAGE=openapitools/openapi-generator-cli:v5.3.0
 OPENAPI_LEGACY_GENERATOR=$(DOCKER) run --user $(UID_GID) --rm -v $(shell pwd):/mnt $(OPENAPI_LEGACY_GENERATOR_IMAGE)
-OPENAPI_GENERATOR_IMAGE=openapitools/openapi-generator-cli:v7.0.0
+OPENAPI_GENERATOR_IMAGE=openapitools/openapi-generator-cli:v6.6.0
```
Comment (Jonathan-Rosenberg) [inline Makefile:17]: ?

File: clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java [hunk 3]
```diff
@@ -776,16 +783,24 @@ public LakeFSFileStatus getFileStatus(Path path) throws IOException {
         ObjectsApi objectsApi = lfsClient.getObjectsApi();
         // get object status on path
         try {
-            ObjectStats objectStat = objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath(), false, false);
-            return convertObjectStatsToFileStatus(objectLoc, objectStat);
+            ObjectsApi.APIstatObjectRequest objectStat = objectsApi.statObject(objectLoc.getRepository(),
+                                                           objectLoc.getRef(),
+                                                           objectLoc.getPath());
+            objectStat = objectStat.userMetadata(false).presign(false);
+            ObjectStats os = objectStat.execute();
```
Comment (Jonathan-Rosenberg) [inline clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java:790]:
```suggestion
            ObjectStats os = objectStat.userMetadata(false).presign(false).execute();
```

File: notes.org
Comment (Jonathan-Rosenberg) [inline notes.org:1]: What's this?

File: project/metals.sbt
Comment (Jonathan-Rosenberg) [inline project/metals.sbt:1]: 🤘 s

File: project/project/metals.sbt
Comment (Jonathan-Rosenberg) [inline project/project/metals.sbt:1]: both `project/metals.sbt` and `project/project/metals.sbt`?

**Review by Jonathan-Rosenberg — CHANGES_REQUESTED** [review]
Looks exhausting....
Thank you!
Some minor things...

**Comment by arielshaqed** [comment]
PTAL... I had to merge `master` into this branch (not rebase) to simplify the large number of merge conflicts.  I hope I fixed everything :-)

File: project/project/metals.sbt
Comment (arielshaqed) [reply project/project/metals.sbt:1]: Both wrong.  Removed.  Will add to my local machine gitignore.

File: project/metals.sbt
Comment (arielshaqed) [reply project/metals.sbt:1]: # 🥈

File: clients/hadoopfs/pom.xml (hunk 1 above)
Comment (arielshaqed) [reply clients/hadoopfs/pom.xml]: This is unfortunately needed until I push a first version of the Java client SDK.

**Review by arielshaqed — COMMENTED** [review]
THANKS!
PTAL...

**Comment by arielshaqed** [comment]
PTAL.  I am downgrading to "Draft" so I do not pull it by mistake.  In order to pull we need to:
* [ ] Release a version of io.lakefs:sdk-client that is _not_ a release candidate.

  A principal issue to resolve is whether we can do that without a full lakeFS release.
* [ ] Use that version of sdk-client in our pom.xml.

File: .github/workflows/esti.yaml [hunk 4]
```diff
@@ -295,6 +295,12 @@ jobs:
           AWS_SECRET_ACCESS_KEY: minio123
           STORAGE_NAMESPACE: s3://test-bucket/data
           REPOSITORY: lakefsfs-contract-test
+
+      - name: Build current lakeFS SDK
+        # TODO(ariels): Remove this in favour of a single lakeFS client version
```
Comment (Jonathan-Rosenberg) [inline .github/workflows/esti.yaml:300]: are we removing this or the legacy version?

File: .github/workflows/esti.yaml [hunk 5]
```diff
@@ -381,6 +387,10 @@ jobs:
           distribution: "adopt-hotspot"
           java-version: "8"
           cache: "sbt"
+      - name: Build current lakeFS SDK
+        # TODO(ariels): Remove this in favour of a single lakeFS client version
```
Comment (Jonathan-Rosenberg) [inline .github/workflows/esti.yaml:391]: same

**Review by Jonathan-Rosenberg — APPROVED** [review]
One question, but looks good

File: .github/workflows/esti.yaml (hunk 4 above)
Comment (arielshaqed) [reply .github/workflows/esti.yaml:300]: Both!  For now...

Later on we will remove the old-style API.  But we need to allow our users to upgrade servers first, then clients.

**Review by arielshaqed — COMMENTED** [review]
THANKS!

I will pull once we release the client with no "RC.0" on it, I guess.

**Comment by arielshaqed** [comment]
Pulling...

**Pull request merged**